core/src/main/java/hudson/Util.java

```diff
@@ -617,6 +620,13 @@ public static String getDigestOf(@Nonnull InputStream source) throws IOException
         */
     }
 
+    // TODO JENKINS-60563 remove MD5 from all usages in Jenkins
+    @SuppressFBWarnings(value = "WEAK_MESSAGE_DIGEST_MD5", justification =
+            "This method should only be used for non-security applications where the MD5 weakness is not a problem.")
+    private static MessageDigest getMd5() throws NoSuchAlgorithmException {
```

Comment:
Good time to deprecate this and all callers?

Reply:
Deprecating this makes sense. I've wondered about deprecating `Util.getDigestOf()`. Would it make sense to change the implementation to SHA-2? The documentation specifies MD5, but it might make sense to change that along with the implementation. It wouldn't be the first JavaDoc in Jenkins that is out-of-date, wrong, or not helpful.

Reply:
I've figured it made more sense to answer that question when engaged in an effort to remove MD5.

Reply:
By the time we fully remove MD5, maybe SHA-4 will be a thing by then! We can defer implementation until then as you suggest. :)